File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> d = new HashMap<>();
        for (int i=0; i<nums.length; i++){
            int val = nums[i];
            int diff = target - nums[i];
            if (d.containsKey(diff)){
                return new int[]{d.get(diff), i};
            }
            d.put(val, i);

        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/majority-element/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d = {}
        n = len(nums)
        for i in nums:
            d[i] = 1 + d.get(i, 0)
        for i in nums:
            if d.get(i) > n//2:
                return i 
        
